sdks/python/apache_beam/io/avroio.py

```diff
@@ -586,30 +591,32 @@ def __init__(self,
     self._schema = schema
     self._codec = codec
 
+  def display_data(self):
+    res = super(_BaseAvroSink, self).display_data()
+    res['codec'] = str(self._codec)
+    res['schema'] = str(self._schema)
+    return res
+
+
+class _AvroSink(_BaseAvroSink):
+  """A sink for avro files using Avro. """
   def open(self, temp_path):
-    file_handle = super(_AvroSink, self).open(temp_path)
+    file_handle = super(_BaseAvroSink, self).open(temp_path)
     return avro.datafile.DataFileWriter(
         file_handle, avro.io.DatumWriter(), self._schema, self._codec)
 
   def write_record(self, writer, value):
     writer.append(value)
 
-  def display_data(self):
-    res = super(_AvroSink, self).display_data()
-    res['codec'] = str(self._codec)
-    res['schema'] = str(self._schema)
-    return res
-
 
-class _FastAvroSink(_AvroSink):
-  """A sink for avro files that uses the `fastavro` library"""
+class _FastAvroSink(_BaseAvroSink):
+  """A sink for avro files using FastAvro. """
   def open(self, temp_path):
-    file_handle = super(_AvroSink, self).open(temp_path)
-    return Writer(file_handle, self._schema.to_json(), self._codec)
+    file_handle = super(_BaseAvroSink, self).open(temp_path)
```

Comment:
super() takes the subclass as first argument. So this should be `super(_FastAvroSink, self)`